### PR TITLE
Added localized navigation components

### PR DESCRIPTION
**Pull request by Fuzznimp**
## Description

This PR aims at localizing our navigation components. This was done in a separate PR because these components are built from hashes, and required more refactoring than just replacing the string with a `formatMessage` call.

I mostly kept the implementation.

I had to declare two `intl`, like for the server side, because the layout is accessed is every pages.

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by frieda-ewlines** [comment]
Task linked: [CU-mv6xrx [feature] - i18n for dynamic nav components](https://app.clickup.com/t/mv6xrx)

File: content/locales/fr.ts [hunk 1]
```diff
@@ -73,4 +77,16 @@ export const fr = {
     profileCreated: "Votre profil à été créé",
     profileUpdated: "Votre profil a été mis à jour",
   },
+  navigation: {
+    home: "Accueil",
+    logins: "Identifiants",
+    security: "Sécurité",
+    createYourProfile: "Créer votre profil",
+    personalInformation: "Informations personelles",
```
Comment (Oluane) [inline content/locales/fr.ts:85]:
```suggestion
    personalInformation: "Informations personnelles",
```

**Review by Oluane — DISMISSED** [review]

**Comment by Fuzznimp** [comment]
@Oluane Following my changed to the profile happy path e2e test, I ran it 2 times, and both succeeded.

File: tests/e2e/profile-happy-path.test.ts [hunk 2]
```diff
@@ -42,7 +43,7 @@ describe.only("Profile Happy path", () => {
   ).includes("staging");
 
   test("it should do Profile and Addresses flows' happy path", async () => {
-    expect.assertions(isStagingEnv ? 29 : 21);
+    expect.assertions(isStagingEnv ? 29 : 19);
```
Comment (Oluane) [inline tests/e2e/profile-happy-path.test.ts:46]:
```suggestion
    expect.assertions(isStagingEnv ? 28 : 19);
```
If I'm not wrong, you delete two expects, one inside a non-staging env case and one that occurs for both cases.

Comment (Fuzznimp) [reply tests/e2e/profile-happy-path.test.ts:46]: Oh you are right, good catch!

**Review by Oluane — APPROVED** [review]

**Pull request merged**